### ISSUE TITLE
Long email addresses are false negatives

### ISSUE THREAD

**Issue by bradvogel** [hapijs/isemail#13]
Why are long auto-reply email addresses flagged as not being an email? These are very common. The two addresses below are from auto-generated Github and Asana emails:

```
brad: nodemailertest npm install isemail
npm WARN package.json nodemailertest@1.0.0 No description
npm WARN package.json nodemailertest@1.0.0 No repository field.
npm WARN package.json nodemailertest@1.0.0 No README data
isemail@2.1.0 node_modules/isemail
brad: nodemailertest node
> var Isemail = require('isemail');
undefined
> Isemail.validate('reply+000c89caea14b0bf53caf17fd0c9c67fce378f51fd1af05492cf000000011292f3d892a169ce07601530@reply.github.com')
false
> Isemail.validate('2.76502047696709.12649470900136.76502049898901.4mtgutesvvbpnqn5wzg13g81kw8cn7xk4wz6nffte6jyjb7hx8@mail.asana.com')
false
> 



```

**Comment by bradvogel** [comment]
This is an important issue because of the reliance on this library by the very popular Nodemailer (https://github.com/andris9/Nodemailer) library. So it's very common for these email addresses to be encountered. See https://github.com/andris9/smtp-connection/issues/32

**Comment by gergoerdosi** [comment]
They are flagged because the local part is too long. RFC 5321 allows only 64 octets:

http://tools.ietf.org/html/rfc5321#section-4.5.3.1.1

**Comment by bradvogel** [comment]
Ok, thanks. Is there a way to configure this library to ignore that specific part of the RFC?

**Comment by skeggse** [comment]
Unfortunately, that particular violation of 5321 assigns the email address a relatively high `errorLevel`, where higher values are worse infractions. This is because the email address is technically invalid per SMTP (as described in RFC 5321), and some parts of the email infrastructure may be unable to accommodate the email address. The only currently implemented method of ignoring errors is by a threshold for `errorLevel`, and since this violation is at such a large value, it's impossible to specifically allow this.

In this case, I don't know there's a lot we can do because Nodemailer shouldn't really allow sending emails with email addresses that might not make it through SMTP nodes. I'd recommend reaching out to GitHub and Asana directly.

**Comment by bradvogel** [comment]
Thanks @skeggse . Makes sense. I'll talk to the maintainer of nodemailer and see if we can work out a bypass.

**Comment by skeggse** [comment]
If you have a better idea for error handling, I'm all ears.

**Comment by skeggse** [comment]
This should be addressed in 4.0.0, either via #162 or #4.
